ios/app/brave_main_delegate.mm

```diff
@@ -21,20 +21,13 @@
 #endif
 
 namespace {
-
-#if defined(OFFICIAL_BUILD)
-// production
-const char kBraveSyncServiceURL[] = "https://sync-v2.brave.com/v2";
-#else
-// For local server development "http://localhost:8295/v2 can also be overriden
-// by switches::kSyncServiceURL
-// dev
-const char kBraveSyncServiceURL[] = "https://sync-v2.brave.software/v2";
-#endif
-
+const char kBraveSyncServiceURL[] = BRAVE_SYNC_ENDPOINT;
 }  // namespace
 
 BraveMainDelegate::BraveMainDelegate() : brave_sync_service_url_(kBraveSyncServiceURL) {
+  // Guarantee that we have a default URL when compiling
+  static_assert(sizeof(kBraveSyncServiceURL) > 0, "Invalid Sync Service URL");
```

Comment:
this is not the best way to handle this, we should have an assert in gn `if (is_official_build) { assert(brave_sync_endpoint != "") }`

Reply:
cc @darkdh

Reply:
that is a good idea to check in gn for official build